docs/source/starter/style_guide.rst

```diff
@@ -176,19 +189,25 @@ However, when using DataParallel, you will need to call forward manually
 ****
 Data
 ****
+
 These are best practices for handling data.
 
 Dataloaders
 ===========
-Lightning uses dataloaders to handle all the data flow through the system. Whenever you structure dataloaders,
+
+Lightning uses :class:`~torch.utils.data.DataLoader` to handle all the data flow through the system. Whenever you structure dataloaders,
 make sure to tune the number of workers for maximum efficiency.
 
-.. warning:: Make sure not to use ddp_spawn with num_workers > 0 or you will bottleneck your code.
+.. warning:: Make sure not to use ``Trainer(strategy="ddp_spawn")`` with ``num_workers>0`` in a DataLoader or you will bottleneck your code.
 
 DataModules
 ===========
-Lightning introduced datamodules. The problem with dataloaders is that sharing full datasets is often still challenging
-because all these questions need to be answered:
+
+The :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is designed as a way of decoupling data-related
+hooks from the :class:`~pytorch_lightning.core.lightning.LightningModule` so you can develop dataset agnostic models. It makes it easy to hot swap different
+datasets with your model, so you can test it and benchmark it across domains. It also makes sharing and reusing the exact data splits and transforms across projects possible.
+
+The problem with dataloaders is that sharing full datasets is often still challenging because all these questions need to be answered:
```

Comment:
This does not longer fit well since you have introduced the concept already above. So the impact of asking these questions is lost. Let's make sure the flow of the intro does not break here.

Reply:
removed the long intro.. I guess the reference is enough.